orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategy.groovy

```diff
@@ -52,6 +52,28 @@ interface ResizeStrategy {
     Integer min
   }
 
+  @Canonical
+  static class Source implements Serializable {
+    Collection<String> zones
+    Collection<String> regions
+    String region
+    String zone
```

Comment:
Ya, this is unfortunate.

I don't think there's a way we can do it explicitly within Jackson (maybe @cfieber knows) ... I've not looked into custom deserializers or anything.

What could be done is a transformation when the pipeline/task first starts that say injects `_location` or `_locations` into the JSON if it ever sees zones/regions/region/zone. 

I say `_location` just to have a key that isn't going to collide with a normal user-provided value.